http-server/src/server.rs

```diff
@@ -301,11 +301,15 @@ impl<M: Middleware> Server<M> {
 					// Run some validation on the http request, then read the body and try to deserialize it into one of
 					// two cases: a single RPC request or a batch of RPC requests.
 					async move {
-						if let Err(e) = access_control_is_valid(&access_control, &request) {
-							return Ok::<_, HyperError>(e);
-						}
+						// Only `POST` and `OPTIONS` methods are allowed.
+						match *request.method() {
+							Method::POST if content_type_is_json(&request) => (),
```

Comment:
perhaps combine with the below to avoid needing an empty match body?

eg:

```
match *request.method() {
    Method::POST if !content_type_is_json(&request) => return Ok::<_, HyperError>(response::unsupported_content_type()),
   // ...
```